### DISCUSSION TITLE
Lidl 368308_2010 battery

### DISCUSSION THREAD

**Discussion by Roxxor91** [Koenkk/zigbee2mqtt#11862] [Devices]
Hi!
The Lidl 368308_2010 thermostat does not expose a battery level, but just a number in mV. If we accumulate some numbers, I guess we can also implement a battery level indicator, right? I already collected some numbers. The display on the device shows a battery symbol with 4 levels/stripes. So here is the data I have right now. Maybe someone can also add data? We could then convert this into battery levels.
- 1410 4 Stripes
- 1350 4 Stripes
- 1320 4 Stripes
- 1130 3 Stripes
- 1120 3 Stripes
- 1100 2 Stripes
- 1060 2 Stripes
- 1050 2 Stripes
- 950   2 Stripes

**Comment by Roxxor91** [comment]
Reminder for the future: To integrate this, it has to be enabled like so:
`meta: {battery: {voltageToPercentage: '3V_2850_3000_log'}},
        exposes: [e.battery(), e.battery_voltage(),`
See eg.g. here [https://github.com/Koenkk/zigbee-herdsman-converters/search?q=voltageToPercentage&type=code](url)
Also we need a custom "profile", to be added here: [https://github.com/Koenkk/zigbee-herdsman-converters/blob/31e062701e50908d6bd3e61bb3e4cd39bdff30c5/lib/utils.js#L141](url)

Also I read the voltage of the batteries. The device with "950"mV, actually has 2 batteries (NiMh) with each 1.33V. So no clue what this number means, or if it is actually correct.

**Reply by martin53757** [reply]
I have two of these devices too and I'd like to see an improvement here. Can I help somehow to make it work?
